bin/main.dart

```diff
@@ -58,58 +87,142 @@ _translateLocalizedFiles(List<String> langCodes, String dirPath) async {
     exit(0);
   }
 
-  Map<String, Map<String, String>> oldLangStrMap = Map();
-  Map<String, Map<String, String>> newLangStrMap = Map();
-
+  // loading keys and strings to langStringMap for each language
+  Map<String, Map<String, String>> langStringMap = Map();
   for (final lang in langCodes) {
     if (!_isSupported(lang)) {
       stdout.writeln('Language code $lang is not supported.');
       continue;
     }
-    oldLangStrMap[lang] = await _loadStrings(lang, dirPath);
+    langStringMap[lang] = await _loadStrings(lang, dirPath);
   }
 
-  var gtr = GoogleTranslator();
-  var updated = false;
-
-  await Future.forEach(oldLangStrMap.entries, (oldLangStrOuterMap) async {
-    String sourceLang = oldLangStrOuterMap.key;
-    Map<String, String> sourceLangMap = oldLangStrOuterMap.value;
-    await Future.forEach(sourceLangMap.entries, (sourceLangMapEntry) async {
-      String sourceKey = sourceLangMapEntry.key;
-      String sourceString = sourceLangMapEntry.value;
-      await Future.forEach(oldLangStrMap.entries, (oldLangStrInnerMap) async {
-        String targetLang = oldLangStrInnerMap.key;
-        Map<String, String> targetLangMap = oldLangStrInnerMap.value;
+  // looking for and collecting strings that exist for one language, but don't for another
+  // we need such a strange structure to translate a bunch of strings at once,
+  // because this is much quicker
+  // Map of <targetLang, sourceLang> to List<key>>
+  Map<Tuple2<String, String>, List<String>> toTranslateMap = Map();
+  langStringMap.forEach((sourceLang, sourceStringMap) {
+    sourceStringMap.forEach((sourceKey, sourceString) {
+      langStringMap.forEach((targetLang, targetStringMap) {
         if (sourceLang != targetLang) {
-          if (sourceLangMap[sourceKey].isNotEmpty &&
-              !targetLangMap.containsKey(sourceKey)) {
-            if (!newLangStrMap.containsKey(targetLang)) {
-              newLangStrMap[targetLang] = Map();
-            }
-            updated = true;
-            newLangStrMap[targetLang][sourceKey] = (await gtr
-                    .translate(sourceString, from: sourceLang, to: targetLang))
-                .text;
+          if (sourceStringMap[sourceKey].isNotEmpty &&
+              !targetStringMap.containsKey(sourceKey)) {
+            var tuple = Tuple2(targetLang, sourceLang);
```

Comment:
```suggestion
            final tuple = Tuple2(targetLang, sourceLang);
```

Reply:
you do not change the value here, so let's keep it final and not var

Reply:
Fixed.